### ISSUE TITLE
[BUG] Menhir internal failure on 5.7 with Dune's release profile

### ISSUE THREAD

**Issue by johnridesabike** [ocsigen/js_of_ocaml#1579]
I have a parser written with Menhir that recently started crashing when I compile with JSOO and the Dune release profile. 

```
$ dune test --profile=release
Internal failure -- please contact the parser generator's developers.
[ 0, [ 248, 'Assert_failure', -11 ], [ 0, 'lib/parser.ml', 1995, 4 ] ]
```

I haven't touched this parser's code in a long time, and it has always worked without a problem, so I was suspicious that a dependency broke something. Sure enough, downgrading to JSOO 5.6 works fine.

Curiously, both 5.6 and 5.7 work with Dune's dev profile. Only 5.7 and the release profile triggers this crash.

I don't have much more information yet, and the fact that it's crashing from Menhir-generated code makes debugging it more of a challenge. I'm honestly not sure what to look for, so let me know if there are any other details I can provide that will help.

**Comment by hhugo** [comment]
Can you try to use the dev version
`opam pin add js_of_ocaml-compiler --dev`

**Comment by johnridesabike** [comment]
I'm still getting the same result after `opam pin add js_of_ocaml-compiler --dev` and `dune clean`. Running the JS code after building with Dune's `--profile release` crashes with the same assert failure. The dev profile build still runs without a problem though.

**Comment by hhugo** [comment]
Can you share code so I can reproduce the issue ?

**Comment by johnridesabike** [comment]
The repository is here if you want to look at it: https://github.com/johnridesabike/acutis

It has a lot of moving parts, so it will take me time to trim it down to a minimal reproducible example though.

```
$ dune test --force --profile=dev # passes
$ dune test --force --profile=release # assertion failure
```

**Comment by hhugo** [comment]
#1580 should fix the issue. Can you try and report ?

**Comment by johnridesabike** [comment]
I see you already merged the change, but I just want to confirm that yes it did fix the error! Thank you.

**Comment by hhugo** [comment]
Fix released inside version `5.7.1`
